### PR TITLE
IGNITE-22618 Remove MvPartitionStorage#rowsCount

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-22618

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java [hunk 1]
```diff
@@ -214,13 +218,11 @@ void testSimpleCommit() {
         storageUpdateHandler.handleUpdate(txUuid, UUID.randomUUID(), partitionId, row2, true, null, null, null, null);
         storageUpdateHandler.handleUpdate(txUuid, UUID.randomUUID(), partitionId, row3, true, null, null, null, null);
 
-        assertEquals(3, storage.rowsCount());
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java:217]: Why?

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandlerTest.java [hunk 2]
```diff
@@ -201,8 +197,6 @@ void testUpdateAllBatchedTryLockFailed() {
 
         storageUpdateHandler.handleUpdateAll(txUuid, rowsToUpdate, partitionId, true, null, null, null);
 
-        assertEquals(3, storage.rowsCount());
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandlerTest.java:204]: Why?

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java:217]: Here is a check that after 3 calls to `StorageUpdateHandler#handleUpdate` only 3 records will be added; I consider such checks useful and controlled paranoid.

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java [hunk 3]
```diff
@@ -231,19 +231,21 @@ private static long[] partitionSizes() {
...
     }
 
-    private static int differences(long[] partitionSizes0, long[] partitionsSizes1) {
+    private static long partitionSize(MvPartitionStorage partitionStorage) {
+        try (PartitionTimestampCursor cursor = partitionStorage.scan(HybridTimestamp.MAX_VALUE)) {
+            return cursor.stream().count();
+        }
+    }
+
+    private static long differences(long[] partitionSizes0, long[] partitionsSizes1) {
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:245]: yes, but IDEA was complaining about the cast

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java:217]: We literally check the same thing on the next line with Mockito: 
```
// We have three writes to the storage.
verify(storage, times(3)).addWrite(any(), any(), any(), anyInt(), anyInt());
```

File: modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexNodeFinishedRwTransactionsCheckerTest.java:245]: Ok

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageCleanupTest.java:217]: Ok.

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**